### PR TITLE
Process different client requests uniquely. 

### LINKED ISSUE #50: Refactor gateway to process requests from web clients
Test web client login via gateway without the Data Access Filter Implementation, this will mimic the current authentication via keycloak that all requests return all data in HAPI without any additional filters. This may request changes in the gateway to uniquely identify web requests and implementing a bypass for data access checker.

### PR DESCRIPTION
**Pull request by lincmba**
Identify client using keycloak roles
 Process different client requests differently
 fixes https://github.com/onaio/fhir-gateway-extension/issues/50

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Resolves [link to issue]

**Engineer Checklist**

- [x] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [x] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [x] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [x] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [x] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: exec/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.smartregister</groupId>
     <artifactId>opensrp-gateway-plugin</artifactId>
-    <version>1.0.14</version>
+    <version>1.0.15</version>
```
Comment (dubdabasoduba) [inline exec/pom.xml:7]: @lincmba we have added new work so let's move to `2.0.0`

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Review by dubdabasoduba — APPROVED** [review]
